cmd/fleet/vuln_process.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/fleetdm/fleet/v4/server/datastore/mysql"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/spf13/cobra"
+)
+
+var (
+	dev               bool
+	devLicense        bool
+	devExpiredLicense bool
+	lockDuration      time.Duration
+)
+
+func createVulnProcessingCmd(configManager config.Manager) *cobra.Command {
+	vulnProcessingCmd := &cobra.Command{
+		Use:   "vuln_processing",
+		Short: "Run the vulnerability processing features of Fleet",
+		Long: `The vuln_processing command is intended for advanced configurations that want to externally manage 
+vulnerability processing. By default the Fleet server command internally manages vulnerability processing via scheduled
+'cron' style jobs, but setting 'vulnerabilities.disable_schedule=true' or 'FLEET_VULNERABILITIES_DISABLE_SCHEDULE=true' 
+will disable it on the server allowing the user configure their own 'cron' mechanism`,
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := configManager.LoadConfig()
+			if dev {
+				applyDevFlags(&cfg)
+			}
+
+			licenseInfo, err := initLicense(cfg, devLicense, devExpiredLicense)
+			if err != nil {
+				initFatal(
+					err,
+					"failed to load license - for help use https://fleetdm.com/contact",
+				)
+			}
+
+			if licenseInfo != nil && licenseInfo.IsPremium() && licenseInfo.IsExpired() {
+				fleet.WriteExpiredLicenseBanner(os.Stderr)
+			}
+
+			ds, err := mysql.New(cfg.Mysql, clock.C)
+			if err != nil {
+				initFatal(err, "creating db connection")
+			}
+
+			// we need to ensure this command isn't running with an out-of-date database
+			status, err := ds.MigrationStatus(cmd.Context())
+			if err != nil {
+				initFatal(err, "retrieving migration status")
+			}
+
+			var migrationError error
+			switch status.StatusCode {
+			case fleet.AllMigrationsCompleted:
+				// only continue if db is considered up-to-date
+			case fleet.NoMigrationsCompleted:
+				migrationError = errors.New("no migrations completed")
+			case fleet.SomeMigrationsCompleted:
+				migrationError = errors.New("partial migrations completed")
+			case fleet.UnknownMigrations:
+				migrationError = errors.New("database migrations incompatible with current version")
+			}
+			if migrationError != nil {
+				initFatal(migrationError, "refusing to continue processing vulnerabilities, database out of sync")
+			}
+
+			ctx := context.Background()
+			ctx = license.NewContext(ctx, licenseInfo)
+			instanceID, err := server.GenerateRandomText(64)
+			if err != nil {
+				initFatal(errors.New("error generating random instance identifier"), "")
+			}
+			// using the same lock name as the cron scheduled version of vuln processing, that way if we fail to obtain the lock
+			// it's most likely due to vulnerabilities.disable_schedule=false but still trying to run external vuln processing command
+			lock, err := ds.Lock(ctx, "cron_vulnerabilities", instanceID, lockDuration)
```

Comment:
I believe the name of the lock is "vulnerabilities", see:https://github.com/fleetdm/fleet/blob/e4a8bf413093ebc5635e7e47376cac0457c9505e/server/fleet/cron_schedules.go#L18